api/server/router/container/backend.go

```diff
@@ -18,7 +18,7 @@ type execBackend interface {
 	ContainerExecCreate(name string, config *types.ExecConfig) (string, error)
 	ContainerExecInspect(id string) (*backend.ExecInspect, error)
 	ContainerExecResize(name string, height, width int) error
-	ContainerExecStart(ctx context.Context, name string, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer) error
+	ContainerExecStart(ctx context.Context, name string, stdin io.Reader, stdout io.Writer, stderr io.Writer) error
```

Comment:
This one seems a bit string.
I'd expect to need to close `stdin` if the exec/container process exits.

Reply:
Yes, that's what I was thinking too, but apparently that's not the case.

I'll take another look to find out why.

Reply:
It looks like the caller is responsible for closing the the reader, which is why this isn't using `ReadCloser`